code/modules/mob/living/simple_animal/friendly/snake.dm

```diff
@@ -58,9 +66,69 @@
                 visible_message("<span class='notice'>[name] consumes [target] in a single gulp!</span>", "<span class='notice'>You consume [target] in a single gulp!</span>")
                 QDEL_NULL(target)
                 adjustBruteLoss(-2)
         else
                 return ..()
+
+/mob/living/simple_animal/hostile/retaliate/poison/snake/asclepius
+    name = "Asclepius's snake"
+    desc = "A mystical snake previously trapped upon the Rod of Asclepius, now freed of its burden. Its bites are rumored to have healing properties."
+    poison_type = /datum/reagent/medicine/omnizine //while using godblood instead of omnizine here would be flavorful, doing that would also allow the snake to pump you with up to 150u of godblood (it has an OD threshold of 150u), which would basically leave you set for the rest of the round
+    gold_core_spawnable = NO_SPAWN //only obtainable from the rod of ascelpius and/or adminbus
+    melee_damage_lower = 0 //do no harm
+    melee_damage_upper = 0
+    friendly_verb_continuous = "bites" //so that it will bite people instead of nuzzle them
+    friendly_verb_simple = "bite"
+    buffed = FALSE //so that you can't use a wumborian fugu gland on it to allow it to break its oath
+    food_type = list(/obj/item/reagent_containers/food/snacks/grown) //it's vegan (or at least trying to be a vegetarian)
+    careful = TRUE //otherwise, NPC asclepius snakes will kill people by ODing them on omnizine
+    attack_same = TRUE //we "attack" our allies to heal them
+
+/mob/living/simple_animal/hostile/retaliate/poison/snake/asclepius/Initialize()
+    . = ..()
+    var/datum/atom_hud/H = GLOB.huds[DATA_HUD_MEDICAL_ADVANCED]
+    H.add_hud_to(src)
+    notify_ghosts("A controllable snake of Asclepius has been created in \the [get_area(src)].", source = src, action = NOTIFY_ORBIT, flashwindow = FALSE, header = "Sentient Snake Created")
+
+/mob/living/simple_animal/hostile/retaliate/poison/snake/asclepius/ListTargets(atom/the_target)
+    . = oview(vision_range, targets_from) //get list of things in vision range
+    var/list/patients = list()
+    for (var/HM in .)
+        //are you a creature?
+        if(isliving(HM))
+            var/mob/living/L = HM
+            //can I heal you?
+            if(L.reagents && (L.health < L.maxHealth)) //yeah, clone damage and such will make the snake be obsessed with one target, but eh, even divine doctors ain't perfect
+                if(!(careful && poison_type.overdose_threshold && L.reagents.has_reagent(poison_type, (poison_type.overdose_threshold - poison_per_bite)))) //if they're already full of our reagent, let's not remain obsessed with them
+                    patients += L
+    return patients
+
+/mob/living/simple_animal/hostile/retaliate/poison/snake/asclepius/tamed()
+    ..()
+    friends.Cut() //we WANT to be able to "attack" our friends so that we can heal them
```

Comment:
```suggestion
/mob/living/simple_animal/hostile/retaliate/poison/snake/asclepius/tamed()
	. = ..()
	friends.Cut() //we WANT to be able to "attack" our friends so that we can heal them
```